src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java

```diff
@@ -885,31 +997,45 @@ public void transientNotificationOnlyNodeWithItemTest() throws NoResponseExcepti
                 leafNode.publish(item);
             });
             assertEquals(StanzaError.Type.MODIFY, e.getStanzaError().getType());
-            assertNotNull(e.getStanzaError().getExtension("item-forbidden", "http://jabber.org/protocol/pubsub#errors"));
-        }
-        finally {
+            assertNotNull(
+                    e.getStanzaError().getExtension("item-forbidden", "http://jabber.org/protocol/pubsub#errors"));
+        } finally {
             pubSubManagerOne.deleteNode(nodename);
         }
     }
 
+    /**
+     * Asserts that one receives a published item that contains an 'offline' message event , after subscribing to a
+     * node.
+     * 
+     * @throws NoResponseException   if there was no response from the remote
+     *                               entity.
+     * @throws XMPPErrorException    if there was an XMPP error returned.
+     * @throws NotConnectedException if the XMPP connection is not connected.
+     * @throws InterruptedException  if the calling thread was interrupted.
+     * @throws ExecutionException    if waiting for the response was interrupted.
+     * @throws PubSubException       if the involved node is not a pubsub node.
+     */
+
     @SmackIntegrationTest
     public void receiveOfflineMessageEventItemTest() throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException, ExecutionException, PubSubException
     {
         final String nodename = "sinttest-receive-published-item-nodename-" + testRunId;
-        // final String needle = "test content " + Math.random();
         LeafNode publisherNode = pubSubManagerOne.createNode(nodename);
         try {
             final Node subscriberNode = pubSubManagerTwo.getNode(nodename);
             final EntityBareJid subscriber = conTwo.getUser().asEntityBareJid();
             subscriberNode.subscribe( subscriber );
-            Message message = new Message();
+            Message message = new Message(subscriber);
+            message.setFrom(conOne.getUser().asEntityBareJid());
             MessageEvent event = new MessageEvent();
             event.setOffline(true);
-            event.setStanzaId(StanzaIdUtil.newStanzaId());
+            String stanzaID = StanzaIdUtil.newStanzaId();
+            event.setStanzaId(stanzaID);
             message.addExtension(event);
             final CompletableFuture<Stanza> result = new CompletableFuture<>();
-            conTwo.addAsyncStanzaListener( result::complete, stanza -> stanza.toXML("").toString().contains("offline"));
-            publisherNode.publish( new PayloadItem<>( GeoLocation.builder().setDescription(message.toXML("").toString()).build() ) );
+            conTwo.addAsyncStanzaListener( result::complete, stanza -> stanza.toXML("").toString().contains(stanzaID));
+            publisherNode.publish( new PayloadItem<SimplePayload>(null,new SimplePayload(message.toXML().toString())));
```

Comment:
You almost got it right.

I think that this 'works', in the sense that you will be able to publish data, and others can subscribe to the node on which items are published, and receive data.

However, the way in which the data is 'wrapped' in an item is unconventional, and is not how the standard is expected to be used.

What is a little bit confusing here is that you want to share data that that's typically not shared via pubsub. I think that you indent to share a `MessageEvent`. 

Instead:
- You are sharing an entire message (of which the MessageEvent is a part)
- The pubsub item should be an XML structure. What you are sharing is an XML structure that contains a text that is escaped XML. That's not quite correct.

It was smart of you to use `SimplePayload`, as that allows you to add arbitrary content to an item that you want to publish.

As you might have figured out, the `PayloadItem` needs a type that extends from `ExtensionElement` that defines the data that is included in the item. `SimplePayload` is a very convenient extension of `ExtensionElement`, as you can add any string to it. The downside of using `SimplePayload` is that it does not give you much structure to work with.

I suggest that instead of using `SimplePayload`, you use another type that also extends from `ExtensionElement`. Hint: `MessageEvent` extends `ExtensionElement`!